ui/pages/confirmations/components/confirm/footer/footer.tsx

```diff
@@ -65,19 +65,28 @@ const ConfirmButton = ({
           onSubmit={onSubmit}
         />
       )}
-      <Button
-        block
-        data-testid="confirm-footer-button"
-        startIconName={
-          hasDangerAlerts ? getIconName(hasUnconfirmedDangerAlerts) : undefined
-        }
-        onClick={hasDangerAlerts ? handleOpenConfirmModal : onSubmit}
-        danger={hasDangerAlerts}
-        size={ButtonSize.Lg}
-        disabled={hasUnconfirmedDangerAlerts ? false : disabled}
-      >
-        {dangerAlerts?.length > 1 ? t('reviewAlerts') : t('confirm')}
-      </Button>
+      {hasDangerAlerts ? (
```

Comment:
This is an interesting one one and probably down to personal preference if it's cleaner to think of it as one conceptual button with alternate data based on the scenario, or two conceptually distinct buttons based on the scenario.

In this example, we're trading branched logic for additional lines and duplicated properties.

Reply:
thanks for pointing this out! agreed and well put

Reply:
I agree to @matthewwalsh0 view here